### PR TITLE
DAOS-3988 tools: Add container delete-acl command

### PR DESCRIPTION
**Pull request by kjacque**
This command and client API function allow a user to
remove an entry from an ACL based on the principal (type
and name). On success, the tool fetches the new ACL and
displays it for confirmation.

- Add the cont delete-acl command to the daos user tool.
  This uses a special format to designate the principal.
  Users may pass in one of the special principals (OWNER@,
  GROUP@, EVERYONE@), or designate a named user or group
  by prefixing it with "u:" or "g:".
- Add daos_cont_delete_acl endpoint to the client API.
- Add new cases for delete to daos_test ACL test.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/container/rpc.h [hunk 1]
```diff
@@ -383,6 +386,17 @@ CRT_RPC_DECLARE(cont_prop_set, DAOS_ISEQ_CONT_PROP_SET, DAOS_OSEQ_CONT_PROP_SET)
 CRT_RPC_DECLARE(cont_acl_update, DAOS_ISEQ_CONT_ACL_UPDATE,
 		DAOS_OSEQ_CONT_ACL_UPDATE)
 
+#define DAOS_ISEQ_CONT_ACL_DELETE	/* input fields */	 \
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/container/rpc.h]: False positive

File: src/include/daos_task.h [hunk 2]
```diff
@@ -304,6 +305,12 @@ typedef struct {
 	struct daos_acl		*acl;
 } daos_cont_update_acl_t;
 
+typedef struct {
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/include/daos_task.h]: Ignoring for consistency with the rest of this file

**Comment by kjacque** [comment]
The Functional_Hardware test failure is a duplicate of [DAOS-4171](https://jira.hpdd.intel.com/browse/DAOS-4171).

File: src/tests/suite/daos_container.c [hunk 3]
```diff
@@ -653,6 +655,25 @@ co_acl(void **state)
...
+	assert_int_equal(rc, 0);
+
+	co_acl_get(arg, exp_acl, exp_owner, exp_owner_grp);
+
+	print_message("Case 5: delete entry no longer in ACL\n");
+
+	/* try deleting same entry again - should be gone */
+	rc = daos_cont_delete_acl(arg->coh, type_to_remove, name_to_remove,
+				  NULL);
+	assert_int_equal(rc, -DER_NONEXIST);
```
Comment (kccain) [inline src/tests/suite/daos_container.c]: one possible addition (minor - only if you need to update something else in the PR), supply DAOS_HDL_INVAL as the first argument to daos_cont_delete_acl() and expect -DER_NO_HDL error returned.

Comment (kjacque) [reply src/tests/suite/daos_container.c]: Good idea - I need to fix a conflict before I can merge, so I'll add this.

File: src/container/rpc.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (kjacque) [reply src/container/rpc.h]: False positive

File: src/include/daos_task.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjacque) [reply src/include/daos_task.h]: Ignoring for consistency with this file's code style

**Comment by kjacque** [comment]
The Functional test failures were both instances of [DAOS-3945](https://jira.hpdd.intel.com/browse/DAOS-3945).

**Comment by kjacque** [comment]
The functional test failure was another incidence of [DAOS-3945](https://jira.hpdd.intel.com/browse/DAOS-3945), not related to this PR.

**Pull request merged**